syntax = "proto3";

message Person {
  int32 id = 1;
  string first_name = 2;
  string last_name = 3;
  string company_name = 4;
}

message PersonList {
  repeated Person persons = 1;
}

message PersonRequest {
  optional int32 id = 1;
}

service PersonService {
  rpc Create(Person) returns (Person);
  rpc Get(PersonRequest) returns (Person);
  rpc GetAll(PersonRequest) returns (PersonList);
}
